pgmpy/inference/continuous_sampling.py

```diff
@@ -0,0 +1,579 @@
+                # Burn-in finished used the last value
+                epsilon = epsilon_bar
+
+        return samples
+
+    def generate_sample(self, theta0, num_adapt, num_samples, Lambda, epsilon=None):
+        """
+        Method returns a generator type object whose each iteration yields a sample
+        using Hamiltonian Monte Carlo
+
+        Parameters
+        ----------
+        theta0: A 1d array type object or a row matrix of shape 1 X d
+                or d X 1.(Will be converted to d X 1)
+                Vector representing values of parameter theta, the starting
+                state in markov chain.
+
+        num_adapt: int
+                The number of interations to run the adaptation of epsilon,
+                after Madapt iterations adaptation will be stopped
+
+        num_samples: int
+                     Number of samples to be generated
+
+        Lambda: int or float
+                Target trajectory length, epsilon * number of steps(L),
+                where L is the number of steps taken per HMC iteration,
+                and epsilon is step size for splitting time method.
+
+        epsilon: float , defaults to None
+                 The step size for descrete time method
+                 If None, then will be choosen suitably
+
+        Returns
+        -------
+        genrator: yielding a numpy.array type object for a sample
+
+        Examples
+        --------
+        >>> from pgmpy.inference import HamiltonianMCda as HMCda
+        >>> from pgmpy.inference import JointGaussianDistribution as JGD, GradLogPDFGaussian as GLPG
+        >>> from pgmpy.inference import LeapFrog
+        >>> import numpy as np
+        >>> mean = np.array([1, 1])
+        >>> cov_matrix = np.array([[1, 0.7], [0.7, 3]])
+        >>> model = JGD(mean, cov_matrix)
+        >>> sampler = HMCda(model=model, grad_log_pdf=GLPG, discretize_time=LeapFrog)
+        >>> gen_samples = sampler.generate_sample(np.array([[1], [1]]), num_adapt=10000,
+        ...                                       num_samples = 10000, Lambda=2, epsilon=None)
+        >>> samples = [sample for sample in gen_samples]
+        >>> samples_array = np.concatenate(samples, axis=1)
+        >>> np.cov(samples_array)
+        array([[ 0.98432155,  0.69517394],
+               [ 0.69517394,  2.95449533]])
+        """
+        # TODO: Proper parameterization
+        if isinstance(theta0, (np.matrix, np.ndarray, list, tuple, set, frozenset)):
+            theta0 = np.array(theta0).flatten()
+            theta0 = np.reshape(theta0, (len(theta0), 1))
+        else:
+            raise TypeError("theta should be a 1d array type object")
+
+        if epsilon is None:
+            epsilon = self._find_resonable_epsilon(theta0)
+
+        if num_adapt <= 1:  # return sample generated using Simple HMC algorithm
+            for sample in HamiltonianMC.generate_sample(self, theta0, num_samples, epsilon, Lambda):
+                yield sample
+            return
```

Comment:
@ankurankan Can you suggest a better solution for this ? Python 2.X doesn't allow to return generator objects.